Services/CloudFoundry/CloudFoundryService.cs

```diff
@@ -115,6 +123,71 @@ public async Task<ConnectResult> ConnectToCFAsync(string targetApiAddress, strin
             }
         }
 
+        private static async Task MatchCliVersionToApiVersion()
+        {
+            SemVersion apiVersion = await cfCliService.GetApiVersion();
+            if (apiVersion == null)
+            {
+                _fileLocatorService.CliVersion = 7;
+                dialogService.DisplayErrorDialog(ccApiVersionUndetectableErrTitle, ccApiVersionUndetectableErrMsg);
+            }
+            else
+            {
+                switch (apiVersion.Major)
```

Comment:
I believe the version class (both the one currently in use and MS's) support CompareTo and boolean operators.  So you could rewrite this ala:
```
if (apiVersion < new Version("2.128.0")
...
else if (apiVersion < new Version("2.150.0")
...
else if (apiVersion < new Version("3.363.0")
...
```

Reply:
Ok perfect, that sounds like all we'd need in order to remove that `Semver` dependency 👍